README.md

```diff
@@ -1,2 +1,233 @@
+protocols from the past in order to create something for the future of
+decentralised applications. We want a “roadmap without potholes” for providing
+stronger privacy assurances than Tor for both senders and receivers, while also
+scaling well and providing short-term message storage for offline users. In
+other words, we accept that a project this large requires a piecemeal approach
+but we shall understand and avoid design dead ends that prevent either scaling
+or rigorous analysis of anonymity properties.
+
+We think mix networks occupy a sweet spot in which nodes run extremely efficiently but rigorous analysis remains possible, if challenging.  There are designs like Tor with more efficient designs, but they cannot provide rigorous anonymity properties.  There are also numerous academic schemes designed to support rigorous analysis, but at extreme sacrifices in efficiency.  We choose a Sphinx-like packet format because it’s efficient, adaptable enough, and has good security proofs.  
+
+We leave actually analyzing our scheme for future work in collaboration with academics.  We shall however use Poison mixing and cover traffic strategy similar to Loopix, which appears analyzable although academic work to do so remains ongoing.  We concede that mix networks impose latency on users, but any faster design would definitely not admit rigorous analysis of anonymity properties, and could not credibly claim to be stronger than Tor.  
+
+We shall tweak Sphinx slightly to accommodate both receiver anonymity and
+short-term message storage simultaneously, which may require updating its
+security proofs eventually. We consider short-term message storage essential to
+user experience and advise against doing it via a second layer protocol.
+
+We foresee the public key infrastructure (PKI) being the ultimate scaling
+bottleneck for all existing anonymity scheme designs. We could avoid this with
+gossip protocols but these enable epistemic attacks. We largely leave this to
+future work, but suggest investigating a verifiable gossip based protocol. We
+shall therefore use an insecure gossip based protocol initially with the hope
+that it meshes best with later designs. If we used a more secure design inspired
+by the Tor consensus then we might assumptions elsewhere that limit scalability
+later.
+
+We know rewards for node operators remains a contentious question in the
+anonymity community with seemingly unforeseeable consequences. We nevertheless
+think rewards represents our best hope for a network large enough to challenge
+today’s centralized providers that operate on surveillance capitalism. We do not
+imagine rewards obliviate the need to steward relay operator culture, possibly
+quite the opposite. Messaging types
+We’re focusing on one-to-one messaging for now.  We actually do require messaging layer crypto, even after all the mix net layers, so expect an Axolotl-like ratchet for this.  
```

Comment:
Is ratchet style necessary at this layer? vs putting it above in a "Conversational Security" layer where it comes to protecting what's _in_ the end user messaging.

Reply:
We need a base layer of encryption when messages cross from the outgoing header to the SURB.  If SURBs are provided from recipient to sender then you can do this with a SURB specific key, but that's an enormous design constraint.  

Axolotl is not that complex, so we should just use it for one-to-one messaging.  We'll use something else for group messaging, probably just sender-keys, but maybe something fancier like asynchronous ratcheting trees.

Yes, I'd consider this another "conventional layer", in that some usages will replace it, but we want normal dApp developers to have something imposed on them at this layer.